contracts/RRUtils.sol

```diff
@@ -152,4 +152,39 @@ library RRUtils {
 
         return false;
     }
+
+    function compareNames(bytes memory self, bytes memory other) internal  returns (int){
+        bool equal = self.equals(other);
+        if(equal){ return 0; }
+
+        uint off = 0;
+        uint otheroff = 0;
+        uint head;
+        uint otherhead;
+        uint len = labelCount(self, 0);
+        uint otherlen = labelCount(other, 0);
+
+        while (!equal) {
```

Comment:
A brief description here would be good - something like "keep removing labels from the front of the name until both names are equal, then compare the last nonequal labels to each other".